### PR TITLE
Adds page parameter for enforcing user foreign key constraint

### PR DESCRIPTION
**Pull request by OneFishTwoFishRedFish-JFish**
## What kind of change does this PR introduce?

feature

## What is the current behavior?

Each page/table is required to have a column with a FK to the auth.users table.

## What is the new behavior?

There is a parameter that toggles the user FK requirement.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by OneFishTwoFishRedFish-JFish** [comment]
I see that this might have some overlap with Issue #23 , so feel free to reject if you have something else in mind.

File: src/utils/config.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export interface Page {
   readonly: boolean; // Whether this page is read-only
   id_col: string; // Name of column that refers to the row or item ID - defaults to 'id'
   user_col: string; // Name of column that refers to the user ID and is a foreign key to auth.users - defaults to 'user'
+  enforce_user_col: boolean;
```
Comment (greentfrapp) [inline src/utils/config.ts:27]: Could you add a short comment to explain the attribute?

**Comment by greentfrapp** [comment]
Hey @OneFishTwoFishRedFish-JFish thanks for the contribution! This does overlap with #23 but I'm happy to merge this into dev first as a short-term fix while we figure out how to best solve that.

**Comment by OneFishTwoFishRedFish-JFish** [comment]
i think that'd be really helpful if it could be awhile before it's addressed properly, but will defer to your judgement. thanks

**Comment by greentfrapp** [comment]
Yup! I'll merge it in once you've added the short comment (see above)

**Comment by greentfrapp** [comment]
LGTM thanks @OneFishTwoFishRedFish-JFish! Merging to dev first and will merge to main after more testing

**Pull request merged**